guppy/src/graph/graph_impl.rs

```diff
@@ -809,27 +818,325 @@ impl DependencyMetadata {
         &self.version_req
     }
 
-    /// Returns true if this is an optional dependency.
+    /// Returns true if this is an optional dependency on the platform `guppy` is running on.
+    ///
+    /// This will also return true if this dependency will never be included on this platform at
+    /// all. To get finer-grained information, use the `enabled` method instead.
     pub fn optional(&self) -> bool {
```

Comment:
Does there need to be an optional_on()?

Reply:
For symmetry reasons, I guess. Will add.

Reply:
Done!